src/server/pkg/metrics/metrics.go

```diff
@@ -1,64 +1,125 @@
 package metrics
 
 import (
+	"fmt"
 	"time"
 
+	"github.com/pachyderm/pachyderm/src/client/pkg/config"
 	"github.com/pachyderm/pachyderm/src/client/pkg/uuid"
 	"github.com/pachyderm/pachyderm/src/client/version"
 	db "github.com/pachyderm/pachyderm/src/server/pfs/db"
 
 	"github.com/dancannon/gorethink"
+	"github.com/segmentio/analytics-go"
 	"go.pedge.io/lion/proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
 	kube "k8s.io/kubernetes/pkg/client/unversioned"
 )
 
-func dbMetrics(dbClient *gorethink.Session, pfsDbName string, ppsDbName string, metrics *Metrics) {
+var globalReporter *Reporter
```

Comment:
Why did make this global? I'm generally very against having global structures since they make a number of things more difficult down the line. Is it really necessary here?